### PR TITLE
Add note about different JDBC drivers x native

### PR DESCRIPTION
**Pull request by gastaldi**

### CODE REVIEW

**Review by machi1990 — APPROVED** [review]
LGTM

**Review by geoand — APPROVED** [review]
Nice note!

File: docs/src/main/asciidoc/datasource.adoc [hunk 1]
```diff
@@ -101,6 +101,13 @@ For instance, when we have a JDBC driver extension for a given kind,
 we resolve the JDBC driver to be used automatically, so you don't need to configure it yourself.
 If you want to use a database kind that is not part of this list, just use `other` or whatever name you prefer.
 
+[NOTE]
+====
+Although it's possible to use any other JDBC driver in JVM mode, it's highly unlikely that it will work when compiling your application to native.
+
+We recommend you to use the existing JDBC Quarkus extensions when possible.
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource.adoc]: I would s/when possible/if you plan to compile your application as a native executable/.

Comment (gsmet) [reply docs/src/main/asciidoc/datasource.adoc]: As for the why: you don't have a choice. You have a database, you need to connect to it. You won't migrate your Oracle database to PostgreSQL if you want to give Quarkus a try. So we need to be very careful about how we are wording things.

**Pull request merged**